emukit/experimental_design/model_based/batch_experimental_design.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+
+from emukit.core.acquisition import Acquisition
+from emukit.core.interfaces import IModel
+from emukit.core.loop import CandidatePointCalculator, LoopState
+from emukit.core.optimization import AcquisitionOptimizer
+
+
+class GreedyBatchPointCalculator(CandidatePointCalculator):
+    """
+    Batch point calculator for use in batch experimental design. This point calculator calculates the first point in
+    the batch then adds this as a fake observation in the model with a Y value equal to the mean prediction. The model
+    is reset with the original data at the end of collecting a batch but if you use a model with a non-deterministic
+    training regime, it may lead to the model being modified.
+    """
+    def __init__(self, model: IModel, acquisition: Acquisition, acquisition_optimizer: AcquisitionOptimizer,
+                 batch_size: int=1):
+        """
+        :param model: Model that is used by the acquisition function
+        :param acquisition: Acquisition to be optimized to find each point in batch
+        :param acquisition_optimizer: Acquisition optimizer that optimizes acquisition function to find each point in batch
+        :param batch_size: Number of points to calculate in batch
+        """
+        self.model = model
+        self.acquisition = acquisition
+        self.acquisition_optimizer = acquisition_optimizer
+        self.batch_size = batch_size
+
+    def compute_next_points(self, loop_state: LoopState) -> np.ndarray:
+        """
+        :param loop_state: Object containing history of the loop
+        :return: 2d array of size (batch_size x input dimensions) of new points to evaluate
+        """
+        new_xs = []
+        original_data = (self.model.X, self.model.Y)
+        for i in range(self.batch_size):
```

Comment:
to emphasize that you deliberately not using `i` you could replace it with `for _ in `

Reply:
Will do